### PR TITLE
chore: remove mock data from syncing

### LINKED ISSUE #15142: Allow user to sync devices
As a user
I want to be able to pair/sync my devices
So that I can use status on multiple devices

### Description

The flow starts here: 
https://www.figma.com/file/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=675%3A177908&t=7tNq8e69gJwkCGMo-4

You can use the old profile and plug the new flow there in a new entry ("Syncing"), under Sync settings

Remove any mock data in current version in place of real data, device names etc

### Out of scope
- how to scan
- New profile design
- different error conditions (we can have a single error for now)

@qfrank @Samyoul @siddarthkay will have all the relevant information for the backend, which is being implemented right now

### PR DESCRIPTION
**Pull request by J-Son89**
fixes: https://github.com/status-im/status-mobile/issues/15142

This pr removes the mock data on the syncing pages and adjusts the UI on that page.

Also adds component spec tests to settings list component which were missing.
Updates settings list component to have variant with Status tags in it. (Have requested designers to update Figma file with this variant too)

To Test:
Sync some devices and see the sync device list - page should look correct as in the designs at:

https://www.figma.com/file/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?type=design&node-id=675-177778&t=JYlgu6dKNjwNthjM-4

This page can be navigated to by: Profile -> Syncing

Functionality should work, except for Pair & Unpair device which will be a follow up issue

Also sign in with syncing (onboarding flow) should be checked
https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=3068-333697&t=ewCQ8tWKGQB1KBqV-4

Also for this pr I had to adjust `Quo/Title` component to remove the preset margin to the component and instead allow it to be passed as a prop via a `container-style` prop. As a result I had to add margin to the pages using this: Enable Biometrics, Enable Notifications, Welcome To Web 3, Syncing In Progress and Syncing Results page. I verified all of these using Figma and everything looks correct in terms of positioning and padding 👍

### CODE REVIEW

**Comment by J-Son89** [comment]
@jo-mut we need to align for this as this might affect the page you recently worked on, the other syncing list page in onboarding

File: src/quo2/components/settings/settings_list/style.cljs
Comment (ilmotta) [inline src/quo2/components/settings/settings_list/style.cljs]: Extra line

File: src/status_im2/contexts/syncing/device/style.cljs
Comment (ilmotta) [inline src/status_im2/contexts/syncing/device/style.cljs]: Extra `\n` at the end

File: src/status_im2/contexts/syncing/syncing_devices_list/view.cljs
Comment (ilmotta) [inline src/status_im2/contexts/syncing/syncing_devices_list/view.cljs]: Extra line?

**Review by ilmotta — APPROVED** [review]
Solid PR, no major comments :)

File: src/quo2/components/settings/settings_list/view.cljs
Comment (ulisesmac) [inline src/quo2/components/settings/settings_list/view.cljs]: Be careful, `:on-press` receives a function, and we are passing `on-press` when it exists or `nil` when it doesn't, and it's the same as before this change.

I mean, before:
```clojure
{:on-press on-press
 ,,,
}
```

Now:


```clojure
{:on-press (when on-press on-press)
 ,,,
}
```

File: src/status_im2/contexts/syncing/device/view.cljs
Comment (ulisesmac) [inline src/status_im2/contexts/syncing/device/view.cljs]: I think it'd be better to change this for `cond->` because this is exactly the use case for cond-> and also using `assoc` instead  of `merge` is cheaper.

To be clear, I'm suggesting the following pattern:
```clojure
(cond-> {:your :map}
  bool-val? (assoc :another-key "another value")
)

```

File: src/status_im2/contexts/syncing/syncing_devices_list/view.cljs
Comment (ulisesmac) [inline src/status_im2/contexts/syncing/syncing_devices_list/view.cljs]: We can erase that empty map

File: src/status_im2/contexts/syncing/syncing_devices_list/view.cljs
Comment (ulisesmac) [inline src/status_im2/contexts/syncing/syncing_devices_list/view.cljs]: Same empty map

File: src/status_im2/contexts/syncing/syncing_devices_list/view.cljs
Comment (ulisesmac) [inline src/status_im2/contexts/syncing/syncing_devices_list/view.cljs]: For `unpaired-devices` you could use `remove` instead.

But if you wish, you could use `group-by` to get both in a single pass, followed by a destructuring, like:

```clojure
{:keys [paired unpaired]} (group-by #(if (:enabled? %) :paired :unpaired)) other-devices)
```

wdyt?

**Review by ulisesmac — APPROVED** [review]

**Review by jo-mut — APPROVED** [review]

File: src/status_im2/contexts/syncing/syncing_devices_list/view.cljs
Comment (J-Son89) [reply src/status_im2/contexts/syncing/syncing_devices_list/view.cljs]: yep, will adjust and also the same for your other comments. Thanks for the review!

**Comment by J-Son89** [comment]
@qoqobolo, @VladimrLitvinenko or @pavloburykh just a bump on this one in case you missed it :)

**Comment by qoqobolo** [comment]
Thanks for the ping @J-Son89!
For better visibility (especially when there are a lot of PRs on the board) and to speed up the process, it's better to move PRs to the "e2e tests" column as soon as it is approved

**Comment by J-Son89** [comment]
> Thanks for the ping @J-Son89!
> 
> For better visibility (especially when there are a lot of PRs on the board) and to speed up the process, it's better to move PRs to the "e2e tests" column as soon as it is approved

Ah I didn't know this, thanks! :)

**Comment by pavloburykh** [comment]
@J-Son89 Thanx for the PR. 

I need to admit that I was not able to test PR properly cause I see no way to generate a list of more than 2 paired devices as I have only 1 Android and 1 IOS physical device. Syncing with emulator does not work, syncing with desktop also results in constant failures. 


Based on what I was able to test there couple of questions/issues. 

### ISSUE 1 Online/offline status of synced devices is not shown

**Expected result:** 

https://www.figma.com/file/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?type=design&node-id=675-177778&t=n5BRrUhiLofqgf1w-0

Syncing for Mobile – Figma 2023-06-07 18-21-57

**Actual result:** 
photo_2023-06-07 18 23 58

**Comment by pavloburykh** [comment]
### ISSUE 2 "Unpair" button name is cut inside the button [Android]

Reproduced on Android 12, Samsung Galaxy A52

On screenshot below you can see "p" letter is cut

photo_2023-06-07 18 27 38

**Comment by pavloburykh** [comment]
### ISSUE 3 Android device is always named "localhost' in the list of synced devices

I am also facing this issue in nightly on Successfull sync screen. But I guess this issue might be adressed in scope of this PR. If not - I will log a folloup for that. 

**Actual result:** 

photo_2023-06-07 18 30 30

**Expected result:** 

Android device name should be parsed. In my case it should be Galaxy A52

photo_2023-06-07 18 32 29

**Comment by pavloburykh** [comment]
### ISSUE 4 Case with syncing same device multiple times after reinstalling app/removing profile

To generate a list of synced devices I started syncing same device multiple times, reinstalling app after each sync. 

**Steps:** 

1. Device A generates Sync code
2. Device B scans QR, proceeds with syncing
3. Device B reinstalls app
4. Decvice A generates another Sync code
5. Device B scans and procceds with syncing once again.  (in my case I performed those steps 3 times)

**Actual results:** 

1. Device B observes 2 empty devices in Not paired block of synced devices list (on screenshot there is 1, but should be 2, I have messed up with screenshots). 
photo_2023-06-07 18 41 06
2. At the same time Device A observes 3 devices in Paired section. 
photo_2023-06-07 18 56 10

**Expected result:** 

1. There should be no devices without names in Not Paired section
2. Paired and Unpaired sections should have same devices in the list on all synced devices. In my case Not paired section is empty on Device A and contains devices on Device B. 
3. I am not sure if it is correct to generate a separate device in the list for the same device which performs multiple sync procedures after app re-install.

**Comment by J-Son89** [comment]
Hi @pavloburykh, thanks for the review!

_ISSUE 1 Online/offline status of synced devices is not shown_
This will be future work, currently we do not have any way of getting this data. It’s a feature for Status Go first. I will create follow up issues for both Status Mobile and Status Go to get these features in.

_ISSUE 2 "Unpair" button name is cut inside the button [Android]_
This is actually an issue with the quo2 button for that variant. I will create a follow up issue for this as it is beyond the scope of this pr.

_ISSUE 3 Android device is always named "localhost' in the list of synced devices_
This pr is purely UI focused and the data etc associated here is out of scope. Better to create a follow up issue for this.


_ISSUE 4 Case with syncing same device multiple times after reinstalling app/removing profile_
this is more of a technical detail, given this pr is UI focused I think it’s better to create this issue separately.   

What do you think?

**Comment by pavloburykh** [comment]
> What do you think?

@J-Son89 ok. Let's go ahaed and merge it. 

I also wanted to confirm that there will be/or already exists a followup regarding syncing list page in onboarding

> @jo-mut we need to align for this as this might affect the page you recently worked on, the other syncing list page in onboarding

**Comment by J-Son89** [comment]
> > What do you think?
> 
> @J-Son89 ok. Let's go ahaed and merge it.
> 
> I also wanted to confirm that there will be/or already exists a followup regarding syncing list page in onboarding
> 
> > @jo-mut we need to align for this as this might affect the page you recently worked on, the other syncing list page in onboarding

@pavloburykh all good with the other page that @jo-mut was working on. We discussed and I checked the other page. they share some common code and it still looks correct after I made the adjustments. 👍 

Thanks for the review, I'll create the follow up issues, also for the pairing/unpairing feature to be created.

**Comment by pavloburykh** [comment]
> We discussed and I checked the other page. they share some common code and it still looks correct after I made the adjustments

@J-Son89 those screens do not look aligned right now. Please have a look at it. 
Below I share an example when different number of devices is displayed during ondording aync sucess sreen and in screen from settings. 

So onbording sync screen seems to be broken. 


https://github.com/status-im/status-mobile/assets/97245802/6e508b26-e11b-4efa-b972-4dfa63217ac6

**Comment by J-Son89** [comment]
@pavloburykh sorry I don't mean that those screens are exactly similar. They use some common features on the page but the overall design is different.

sign in via syncing designs:
https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?type=design&node-id=3068-333697&t=2gcoJD65Ihv88b4J-4


in profile syncing designs:
https://www.figma.com/file/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?type=design&node-id=675-177633&t=ftnUpiH3ryeoOhVf-4

**Comment by pavloburykh** [comment]
> @pavloburykh sorry I don't mean that those screens are exactly similar. They use some common features on the page but the overall design is different.

thanx for the answer @J-Son89 But I still do not understand why sign in via syncing does not show same list of devices as profile syncing screen is showing. I beleive this data should be the same. But sign in via syncing screen shows only 2 devices (current device and another paired device) and not showing rest of the devices.

**Comment by pavloburykh** [comment]
Agreed with @J-Son89 not to block the PR as above behaviour is highlikely expected and not affected by the PR. If there will be any need we will log separate issue for that.

**Pull request merged**
